### ISSUE TITLE
SW Renderer fails to render on some Maemo Leste targets

### ISSUE THREAD

**Issue by buzztiaan** [diasurgical/devilutionX#5405]
### Operating System

Other (please specify)

### DevilutionX version

Other (please specify version number)

### Describe

Current git, this is a continuation of https://github.com/diasurgical/devilutionX/issues/5403 

The software renderer fails to render anything on Droid 4 on disabling upscale and having a less than native resolution.
Yet, on the same libsdl2 version ( https://github.com/maemo-leste-upstream-forks/libsdl2 ) on the pinephone, there's no issue whatsoever.

### To Reproduce

edit diablo.ini 

```
[Graphics]
Width=640
Height=480
Fullscreen=1
Fit to Screen=0
Upscale=0
```

Notice that stderr no longer mentions the following :+1: 

`INFO: Created renderer: opengles2`

This is the only confirmation i could find that hw scaler is used.

The game itself seems to run just fine, even catching touch input? And on switching or closing window, i can briefly see a single frame in our compositing window manager ( https://github.com/maemo-leste/hildon-desktop ) but nothing when the window is fullscreen.

This might totally because of our ancient libsdl2 or the weird stuff we did to it.

**Comment by buzztiaan** [comment]
Nokia N900 gives the same results as the Droid 4 , no screen when fullscreen , maybe a frame when switching/closing, touch does seem to arrive. Sound also works fine.

**Comment by StephenCWills** [comment]
I guess it would be helpful to know which SDL video driver is being used on the system. You can add `LogDebug("Video driver: {}", SDL_GetCurrentVideoDriver())` after the call to `SDL_Init()` in `display.cpp`.

https://github.com/diasurgical/devilutionX/blob/c80fcf42c899171d2d27971baacf4a0010730810/Source/utils/display.cpp#L275-L277

If you feel like really diving into this, you could also try checking which drivers are available with `SDL_GetNumVideoDrivers()` and `SDL_GetVideoDriver()`, then maybe even trying different video drivers by changing the `SDL_VIDEODRIVER` environment variable.

**Comment by buzztiaan** [comment]
Totally fine doing so (tomorrow perhaps), but our libsdl2 fork might soon-ish disappear so it might all be for nothing ;)

**Comment by StephenCWills** [comment]
You mean you won't be supporting SDL2 on Maemo Leste, or you simply won't have a fork of it? I noticed that the fork doesn't actually make all that many changes from `2.0.14+dfsg2`.

https://github.com/maemo-leste-upstream-forks/libsdl2/compare/915a939...f038a5c

**Comment by AJenbo** [comment]
You can also try SDL1.2, it only has the SW path no upscaling. Of-cause it also don't have support for touch input so it might be a bit limiting for some devices.

**Comment by buzztiaan** [comment]
There's a new libsdl2 coming soon in Maemo Leste, either we no longer need our own fork, or we pull from upstream until devuan catches up (if at all).

I don't think we will want a version without touch :) The lack of VirtualGamepad on non-multitouch devices is already making me see if i can force it enabled, just for (albeit a bit more challenging) consistency :)

**Comment by AJenbo** [comment]
i was suggesting it to test if it's software render path works as it should, to let us get closer to the source of the issue.

**Comment by qndel** [comment]
@buzztiaan `current git` - there's no such a thing

**Comment by buzztiaan** [comment]
> @buzztiaan `current git` - there's no such a thing

You are correct, this was ment to refer to https://github.com/diasurgical/devilutionX/commit/22adb0a93f1ad53b4fb064ff45a203ce822b8fa7
